numpy/polynomial/polynomial.py

```diff
@@ -776,7 +776,11 @@ def polyval(x, c, tensor=True):
 
     c0 = c[-1] + x*0
     for i in range(2, len(c) + 1):
-        c0 = c[-i] + c0*x
+        # separate out the Horner operation c0 = c[-i] + c0*x,
```

Comment:
I'm a bit concerned here that `c0` must maintain its shape. The `c[-i]` may be a row and `x` can be an arbitrary array. It could be that the initialization of `c0` takes care of that, but it needs thinking about.